lib/miq_expression/count_field.rb

```diff
@@ -4,6 +4,10 @@ class MiqExpression::CountField < MiqExpression::Target
 \.(?<associations>[a-z_\.]+)
 /x
 
+  def self.match(field)
+    self::REGEX.match(field)&.named_captures&.symbolize_keys
+  end
```

Comment:
Minor, but I original had this in the base class MiqExpression::Target, so it could be shared with MiqExpression::Tag, but I felt it was clearer if each one had it's own, considering Field is different.  I expect we will want to change Tag's matcher in a similar way as Field, so this puts the method in place.  If you'd rather I move to the base class though, I can.

Reply:
I was going to say we have it in count_field, field, and tag... do they also have this problem?  Any idea if tests cover the behavior?

Reply:
I tested those and they do not seem to have the same problem, but I want to work on those in separate PRs (another reason I broke this out into the separate classes)